### PR TITLE
Create compiler test variables, and add support for AppleClang (macOS)

### PR DESCRIPTION
**Pull request by johnmcfarlane**

### CODE REVIEW

File: src/test/CMakeLists.txt [hunk 1]
```diff
@@ -277,8 +277,8 @@ add_dependencies(Test-glm Glm)
 
 if(Boost_FOUND AND Boost_VERSION GREATER 106099 AND EXCEPTIONS)
     # Boost.SIMD has problems building with gcc-7.x
-    if(NOT ${CMAKE_CXX_COMPILER_ID} STREQUAL GNU OR CMAKE_CXX_COMPILER_VERSION VERSION_LESS "7.0.0")
-        if (${CMAKE_CXX_COMPILER_ID} STREQUAL Clang OR ${CMAKE_CXX_COMPILER_ID} STREQUAL GNU)
+    if(IS_GCC_FAMILY OR CMAKE_CXX_COMPILER_VERSION VERSION_LESS "7.0.0")
```
Comment (johnmcfarlane) [inline src/test/CMakeLists.txt]: This appears to reverse some logic.
```suggestion
    if(NOT IS_GCC_FAMILY OR CMAKE_CXX_COMPILER_VERSION VERSION_LESS "7.0.0")
```

File: src/test/CMakeLists.txt [hunk 2]
```diff
@@ -296,13 +296,13 @@ if(Boost_FOUND AND Boost_VERSION GREATER 106099 AND EXCEPTIONS)
 
         ExternalProject_Get_Property(BoostSimd source_dir)
         target_include_directories("${target}" PRIVATE SYSTEM "${source_dir}/include")
-    endif(NOT ${CMAKE_CXX_COMPILER_ID} STREQUAL GNU OR CMAKE_CXX_COMPILER_VERSION VERSION_LESS "7.0.0")
+    endif(IS_GCC_FAMILY OR CMAKE_CXX_COMPILER_VERSION VERSION_LESS "7.0.0")
```
Comment (johnmcfarlane) [inline src/test/CMakeLists.txt]: This also:
```suggestion
    endif(NOT IS_GCC_FAMILY OR CMAKE_CXX_COMPILER_VERSION VERSION_LESS "7.0.0")
```

**Comment by johnmcfarlane** [comment]
That did the trick. Thanks!

**Comment by torfinnberset** [comment]
Ah sorry, got a bit too eager on the refactoring. Glad you got it working 👍

**Pull request merged**